test/controllers/ViewAppealDetailsControllerSpec.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.SpecBase
+import helpers.SessionAnswersHelper
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, when}
+import play.api.mvc.Result
+import play.api.test.Helpers._
+import services.SessionService
+import testUtils.AuthTestModels
+import uk.gov.hmrc.auth.core.{AffinityGroup, Enrolments}
+import uk.gov.hmrc.auth.core.retrieve.{Retrieval, ~}
+import views.html.ViewAppealDetailsPage
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ViewAppealDetailsControllerSpec extends SpecBase {
+  val viewAppealDetailsPage: ViewAppealDetailsPage = injector.instanceOf[ViewAppealDetailsPage]
+  val sessionAnswersHelper: SessionAnswersHelper = injector.instanceOf[SessionAnswersHelper]
+  val sessionService: SessionService = injector.instanceOf[SessionService]
+  val ec: ExecutionContext = injector.instanceOf[ExecutionContext]
+
+  class Setup(authResult: Future[~[Option[AffinityGroup], Enrolments]]) {
+    reset(mockAuthConnector)
+    reset(mockSessionService)
+
+    when(mockAuthConnector.authorise[~[Option[AffinityGroup], Enrolments]](
+      any(), any[Retrieval[~[Option[AffinityGroup], Enrolments]]]())(
+      any(), any())
+    ).thenReturn(authResult)
+
+  val controller = new ViewAppealDetailsController(viewAppealDetailsPage, sessionAnswersHelper, sessionService)(mcc, appConfig, authPredicate, ec)
+
+  "ViewAppealDetailsController" should {
+
+    "onPageLoad" when {
+      "the user is authorised" must {
+        "return 200 (OK) and the correct view" in new Setup(AuthTestModels.successfulAuthResult) {
+            when(mockSessionService.getUserAnswers(any())).thenReturn(Future.successful(Some(userAnswers(correctUserAnswers))))
+            val result: Future[Result] = controller.onPageLoad()(userRequestWithCorrectKeys)
+            status(result) shouldBe OK
+          }
+
+        "return 303 (SEE_OTHER) when they do not have user answers" in new Setup(AuthTestModels.successfulAuthResult) {
+            when(mockSessionService.getUserAnswers(any())).thenReturn(Future.successful(None))
+            val result: Future[Result] = controller.onPageLoad()(userRequestWithCorrectKeys)
+            status(result) shouldBe SEE_OTHER
```

Comment:
It is worth checking the redirect location as we can't be certain whether they were redirected because they weren't authenticated or some other reason